src/python/WMComponent/JobAccountant/AccountantWorker.py

```diff
@@ -474,38 +475,30 @@ def handleJob(self, jobID, fwkJobReport):
                 pass
             else:
-                failJob = True
-                if jobType in ["Processing", "Production"]:
-                    cmsRunSteps = 0
-                    for step in fwkJobReport.listSteps():
-                        if step.startswith("cmsRun"):
-                            cmsRunSteps += 1
-                    if cmsRunSteps > 1:
-                        failJob = False
-
-                if failJob:
-                    jobSuccess = False
-                    logging.error("Job %d , list of expected outputModules does not match job report, failing job",
-                                  jobID)
-                    logging.debug("Job %d , expected outputModules %s", jobID, sorted(outputMap.keys()))
-                    logging.debug("Job %d , fwjr outputModules %s", jobID, sorted(outputModules))
-                    fileList = fwkJobReport.getAllFilesFromStep(step='logArch1')
+                # Production and Processing filesets do not consider the logArchive step
+                if set(outputMap) == (outputModules - {"logArchive"}):
+                    pass
                 else:
-                    logging.warning(
-                        "Job %d , list of expected outputModules does not match job report, accepted for multi-step CMSSW job",
-                        jobID)
-            # Make sure every file has a valid location
-            # see https://github.com/dmwm/WMCore/issues/9353
-            for fwjrFile in fileList:
-                # T0 has analysis file without any location, see:
-                # https://github.com/dmwm/WMCore/issues/9497
-                if not fwjrFile.get("locations") and fwjrFile.get("lfn", "").endswith(".root"):
-                    logging.warning("The following file doesn't have any location: %s", fwjrFile)
                     jobSuccess = False
-                    break
+                    errMsg = f"Job {jobID}, expected output modules: {sorted(outputMap.keys())}, "
+                    errMsg += f"but has FWJR output modules: {sorted(outputModules)}"
+                    logging.error(errMsg)
+                    # override file list by the logArch1 output only
+                    fileList = fwkJobReport.getAllFilesFromStep(step='logArch1')
         else:
             fileList = fwkJobReport.getAllFilesFromStep(step='logArch1')
 
+        # Make sure every file has a valid location
+        # see https://github.com/dmwm/WMCore/issues/9353
+        for idx, fwjrFile in enumerate(fileList):
+            # T0 has analysis file without any location, see:
+            # https://github.com/dmwm/WMCore/issues/9497
+            if not fwjrFile.get("locations") and fwjrFile.get("lfn", "").endswith(".root"):
+                logging.warning("The following file does not have any location: %s", fwjrFile)
+                jobSuccess = False
+                # and remove the broken file, otherwise it causes stuckness in WMBS
+                fileList.pop(idx)
```

Comment:
it is dangerous practice to loop through container and delete elements from it within a loop. Instead, you  need to create new list with valid files and use it instead of original `fileList`.